components/camel-debezium/src/main/java/org/apache/camel/component/debezium/configuration/EmbeddedDebeziumConfiguration.java

```diff
@@ -0,0 +1,328 @@
+        addPropertyIfNotNull(configBuilder, EmbeddedEngine.OFFSET_STORAGE_KAFKA_TOPIC, offsetStorageTopic);
+        addPropertyIfNotNull(configBuilder, EmbeddedEngine.OFFSET_STORAGE_KAFKA_PARTITIONS,
+                offsetStoragePartitions);
+        addPropertyIfNotNull(configBuilder, EmbeddedEngine.OFFSET_STORAGE_KAFKA_REPLICATION_FACTOR,
+                offsetStorageReplicationFactor);
+        addPropertyIfNotNull(configBuilder, EmbeddedEngine.OFFSET_COMMIT_POLICY, offsetCommitPolicy);
+        addPropertyIfNotNull(configBuilder, EmbeddedEngine.OFFSET_FLUSH_INTERVAL_MS, offsetFlushIntervalMs);
+        addPropertyIfNotNull(configBuilder, EmbeddedEngine.OFFSET_COMMIT_TIMEOUT_MS, offsetCommitTimeoutMs);
+
+        if (internalKeyConverter != null && internalValueConverter != null) {
+            configBuilder.with("internal.key.converter", internalKeyConverter);
+            configBuilder.with("internal.value.converter", internalValueConverter);
+        }
+
+        return configBuilder.build();
+    }
+
+    protected static <T> void addPropertyIfNotNull(final Configuration.Builder configBuilder,
+                                                   final Field field, final T value) {
+        if (value != null) {
+            configBuilder.with(field, value);
+        }
+    }
+
+    /**
+     * Validate all configurations defined and return {@link ConfigurationValidation} instance which contains the validation results
+     * @return {@link ConfigurationValidation}
+     */
+    public ConfigurationValidation validateConfiguration() {
+        final ConfigurationValidation embeddedEngineValidation = validateDebeziumEmbeddedEngineConfiguration();
+        // only if embeddedEngineValidation is true, we check the connector validation
+        if (embeddedEngineValidation.isValid()) {
+            final ConfigurationValidation connectorValidation = validateConnectorConfiguration();
+
+            ObjectHelper.notNull(connectorValidation, "validateConnectorConfiguration");
+
+            return connectorValidation;
+        }
+        return embeddedEngineValidation;
+    }
+
+    private ConfigurationValidation validateDebeziumEmbeddedEngineConfiguration() {
+        if (isFieldValueNotSet(name)) {
+            return ConfigurationValidation.notValid("Required field 'name' must be set.");
+        }
+        // check for offsetStorageFileName
+        if (offsetStorage.equals(DebeziumConstants.DEFAULT_OFFSET_STORAGE)
+                && isFieldValueNotSet(offsetStorageFileName)) {
+            return ConfigurationValidation.notValid(String
+                    .format("Required field 'offsetStorageFileName' must be set since 'offsetStorage' is set to '%s'",
+                            DebeziumConstants.DEFAULT_OFFSET_STORAGE));
+        }
+        return ConfigurationValidation.valid();
+    }
+
+    protected static boolean isFieldValueNotSet(final Object field) {
+        return field == null || ObjectHelper.isEmpty(field);
```

Comment:
`ObjectHelper#isEmpty` already contains null check.
```suggestion
        return ObjectHelper.isEmpty(field);
```